### PR TITLE
Drop use of build_sphinx distutils command

### PR DESCRIPTION
**Pull request by stephenfin**
Drop use of the `build_sphinx` distutils command, which was dropped in Sphinx 7.x. distutils itself is deprecated and will be removed in Python 3.12.

While we're here, we also fix up some other issues with tox and the setup files in general. The commit messages hold more information about this.

### CODE REVIEW

**Comment by stephenfin** [comment]
Depends on #1329.

**Comment by tobias-urdin** [comment]
@stephenfin thanks, this all looks good to me except for removal of `docs-gnocchi-web` tox env. I think that's used somehow to roll out https://gnocchi.osci.io but I don't have access to that so I'm not sure. It's also building that but ignoring warnings to not fail I guess(?).

Can we leave that in and fix the conflict and I'll pull this in asap to unblock sphinx7, thanks!

**Comment by stephenfin** [comment]
> @stephenfin thanks, this all looks good to me except for removal of `docs-gnocchi-web` tox env. I think that's used somehow to roll out [gnocchi.osci.io](https://gnocchi.osci.io) but I don't have access to that so I'm not sure. It's also building that but ignoring warnings to not fail I guess(?).
> 
> Can we leave that in and fix the conflict and I'll pull this in asap to unblock sphinx7, thanks!

Done. I think the failure is unrelated but I can't rerun it to check.

File: .github/workflows/gnocchi.yml [hunk 1]
```diff
@@ -21,7 +21,6 @@ jobs:
       matrix:
         env:
           - docs
-          - docs-gnocchi-web
```
Comment (tobias-urdin) [inline .github/workflows/gnocchi.yml:24]: Please also add back this

File: .mergify.yml [hunk 2]
```diff
@@ -2,7 +2,6 @@ queue_rules:
   - name: default
     conditions:
     - check-success=doc (docs)
-    - check-success=doc (docs-gnocchi-web)
```
Comment (tobias-urdin) [inline .mergify.yml:5]: And these two

**Review by tobias-urdin — CHANGES_REQUESTED** [review]

**Comment by tobias-urdin** [comment]
Figured out it's handled from here: https://gitlab.com/osci/community-cage-infra-ansible/-/blob/master/roles/gnocchi_builder/templates/gnocchi_builder

File: .mergify.yml (hunk 2 above)
Comment (stephenfin) [reply .mergify.yml]: Done

File: .github/workflows/gnocchi.yml (hunk 1 above)
Comment (stephenfin) [reply .github/workflows/gnocchi.yml]: Done

**Comment by tobias-urdin** [comment]
Sorry for all the resistance @stephenfin – please rebase after revert https://github.com/gnocchixyz/gnocchi/pull/1343

**Comment by stephenfin** [comment]
@tobias-urdin Anything else needed here or can we merge this (I'd like to get it off my TODO list :smile:)

**Comment by tobias-urdin** [comment]
@stephenfin Thanks!

**Pull request merged**